client/export_test.go

```diff
@@ -26,19 +27,34 @@ var (
 )
 
 func (client *Client) SetDoer(d doer) {
-	client.doer = d
+	client.Requester().(*DefaultRequester).doer = d
 }
 
 func (client *Client) Do(method, path string, query url.Values, body io.Reader, v interface{}) error {
-	return client.do(method, path, query, nil, body, v)
+	resp, err := client.Requester().Do(context.Background(), &RequestOptions{
+		Type:    RawRequest,
+		Method:  method,
+		Path:    path,
+		Query:   query,
+		Headers: nil,
+		Body:    body,
+	})
+	if err != nil {
+		return err
+	}
+	err = decodeInto(resp.Body, v)
```

Comment:
Why are we calling the private function here?  Shouldn't this be using the public API?

Reply:
I've clarified this in the TODO comment i've added above:

The test framework do not generate a response of type ```client.response```, which means that we cannot use ```SyncRequest``` or ```AsyncRequest``` as is today, without changing the test suite. These request types assume a client.response can be unmarshalled as the first step.

I am happy to enhance everything to better utilise the new Requester API, but I propose we do this as a followup task.

Reply:
Will address in https://github.com/canonical/pebble/issues/315

Reply:
Agreed to postpone this work.